File: Week14/groovypark/1_solution.py
def solution(participant, completion):
    dic = {}
    for person in participant:
        dic[person] = 0
    for person in participant:
        dic[person] += 1
    for person in completion:
        dic[person] -= 1
    for person in dic:
        if dic[person] == 1:
            return person

print(solution(	["mislav", "stanko", "mislav", "ana"], ["stanko", "ana", "mislav"]))

File: Week16/groovypark/1_solution.js
function solution(participant, completion) {
  let hash = new Map();
  for (const p of completion) {
    if (!hash.has(p)) {
      hash.set(p, 1);
    } else {
      hash.set(p, hash.get(p) + 1)
    }
  }

  for (const p of participant) {
    if (!(hash.has(p))) {
      return p;
    } else {
      if (hash.get(p) === 0) {
        return p;
      } else {
        hash.set(p, hash.get(p) - 1);
      }
    }
  }
}

console.log(solution(["kiki", "kiki", "eden"], ["eden", "kiki"]));